maas.yml

```diff
@@ -77,6 +77,12 @@ runcmd:
 - maas admin maas set-config name=upstream_dns value=8.8.8.8
 # Add LXD as a VM host for MAAS
 - maas admin vm-hosts create  password=password  type=lxd power_address=https://${IP_ADDRESS}:8443 project=maas
+# Automatically create and add ssh keys to MAAS
+- ssh-keygen -q -t rsa -N "" -f "/home/ubuntu/.ssh/id_rsa"
+- chown ubuntu:ubuntu /home/ubuntu/.ssh/id_rsa /home/ubuntu/.ssh/id_rsa.pub
+- chmod 600 /home/ubuntu/.ssh/id_rsa
+- chmod 644 /home/ubuntu/.ssh/id_rsa.pub
```

Comment:
you can do this in one step - create the ssh key in a known place i.e. PATH_TO_KEYS which isn't `/home/ubuntu/.ssh` then:

```
install -o ubuntu -g ubuntu -m 600 $PATH_TO_KEYS/id_rsa /home/ubuntu/.ssh/
install -o ubuntu -g ubuntu -m 640 $PATH_TO_KEYS/id_rsa.pub /home/ubuntu/.ssh/
```